back/boxtribute_server/business_logic/metrics/crud.py

```diff
@@ -121,3 +124,49 @@ def number_of_created_records_between(model, start, end):
         .where((model.created_on >= start) & (model.created_on <= end))
         .count()
     )
+
+
+def family_heads_edited_last_year():
+    cutoff_datetime = datetime.today() - timedelta(days=365)
+    cutoff_date = cutoff_datetime.date()
+
+    edited_family_heads = (
+        DbChangeHistory.select(DbChangeHistory.record_id)
+        .where(
+            (DbChangeHistory.table_name == "people")
+            & (DbChangeHistory.record_id.in_(unique_family_head_ids()))
+            & (DbChangeHistory.change_date >= cutoff_date)
+        )
+        .distinct()
+    )
+
+    edited_family_head_id_list = [row.record_id for row in edited_family_heads]
+    return edited_family_head_id_list
+
+
+def family_heads_in_transaction_last_year():
+    cutoff_datetime = datetime.today() - timedelta(days=365)
+    cutoff_date = cutoff_datetime.date()
+
+    family_head_with_transaction = (
+        Transaction.select(Transaction.beneficiary)
+        .where(
+            (Transaction.beneficiary.in_(unique_family_head_ids()))
```

Comment:
same as above, please join with the Beneficiary model instead of using the subquery